### PR TITLE
Update gen_artifacts_toml.jl

### PR DESCRIPTION
**Pull request by amontoison**
@dpo

### CODE REVIEW

File: utils/gen_artifacts_toml.jl [hunk 1]
```diff
@@ -20,16 +19,34 @@ for format ∈ formats
...
-        "$(matrix.group)/$(matrix.name).$(format)",
-        url,
-        lazy = true,
-        force = true,
-      )
+      problem_hash = artifact_hash("$(matrix.group)/$(matrix.name).$(format)", artifacts_toml)
+      if problem_hash === nothing || !artifact_exists(problem_hash)
+        problem_hash = create_artifact() do artifact_dir
+          download("$url", joinpath(artifact_dir, sscm_file))
+        end
```
Comment (amontoison) [inline utils/gen_artifacts_toml.jl:35]: @AntoninKns
Do you if we should generate a `git-tree-sha1` like this or with https://github.com/simeonschaub/ArtifactUtils.jl/blob/main/src/ArtifactUtils.jl#L52-L57 ?

File: utils/gen_artifacts_toml.jl [hunk 2]
```diff
@@ -20,16 +19,34 @@ for format ∈ formats
   global k
   for matrix ∈ eachrow(matrices)
     k += 1
-    url = "$ssmc_url/$format/$(matrix.group)/$(matrix.name).tar.gz"
+    if format == "mat"
+      sscm_file = "$(matrix.name).mat"
+      url = "$ssmc_url/$format/$(matrix.group)/$sscm_file"
+    else
+      sscm_file = "$(matrix.name).tar.gz"
+      url = "$ssmc_url/$format/$(matrix.group)/$sscm_file"
```
Comment (dpo) [inline utils/gen_artifacts_toml.jl:27]: You could set `url` after the `if` and use `ssmc_file = format == "mat" ? "$(matrix.name).mat" : "$(matrix.name).tar.gz"`

**Comment by dpo** [comment]
`mat` files are not tested. If I update `ssmc_formats`, I get
```
 Downloading artifact: Pajek/Stranke94.mat
[ Info: No hash cache found
[ Info: Calculated hash 48b15e4f1e2d3c4dd1ecde9eb77524a43ea80808467e784e1be8462134b72679 for file /var/folders/rq/p5nq9tv17p5drlk49755jjz80000gn/T/jl_D3cOep-download.gz
[ Info: Unpacking /var/folders/rq/p5nq9tv17p5drlk49755jjz80000gn/T/jl_D3cOep-download.gz into /Users/dpo/.julia/artifacts/jl_bZskMr...
ERROR: /var/folders/rq/p5nq9tv17p5drlk49755jjz80000gn/T/jl_D3cOep-download.gz
/var/folders/rq/p5nq9tv17p5drlk49755jjz80000gn/T/jl_D3cOep-download.gz
Open ERROR: Can not open the file as [gzip] archive


ERRORS:
  Downloaded artifact: Pajek/Stranke94.mat
fetch test: Error During Test at /Users/dpo/dev/JSO/SuiteSparseMatrixCollection.jl/test/runtests.jl:5
  Got exception outside of a @test
  Unable to automatically install 'Pajek/Stranke94.mat' from '/Users/dpo/dev/JSO/SuiteSparseMatrixCollection.jl/src/../Artifacts.toml'
```

**Comment by dpo** [comment]
https://github.com/ScottKolo/suitesparse-matrix-collection-website/issues/25

**Pull request closed without merging**